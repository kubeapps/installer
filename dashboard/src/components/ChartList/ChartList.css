.ChartList__error_nocharts {
  padding-left: 1em;
  font-style: italic;
}
