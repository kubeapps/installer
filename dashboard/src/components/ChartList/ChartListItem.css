.ChartListItem {
  color: #1c2b39;
}

.ChartListItem__content {
}

.ChartListItem__content__title {
  color: #1c2b39;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  font-weight: bold;
  margin: 0;
  width: 100%;
}

.ChartListItem__content__info {
  display: flex;
  justify-content: space-between;
}

.ChartListItem__content__info_repo {
  background-color: #1c2b39;
  border-radius: 16px;
  color: #fff;
  display: inline-block;
}

.ChartListItem__content__info_repo.stable {
  background-color: #1598cb;
}

.ChartListItem__content__info_repo.incubator {
  background-color: #f58220;
}
