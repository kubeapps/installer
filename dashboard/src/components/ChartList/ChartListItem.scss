.ChartListItem {
  color: #1c2b39;
}

.ChartListItem__content__title {
  margin: 0;
  width: 100%;
  font-weight: bold;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  color: #1c2b39;
}

.ChartListItem__content__info {
  display: flex;
  justify-content: space-between;
}

.ChartListItem__content__info_repo {
  display: inline-block;
  color: #fff;
  border-radius: 16px;
  background-color: #1c2b39;
}

.ChartListItem__content__info_other {
  display: inline-block;
  color: rgb(0, 0, 0);
  margin-left: 0.3em;
  border-radius: 16px;
  background-color: #1598cb;
}

.DELETED {
  background-color: #c5c5c5;
}

.FAILED {
  background-color: #cb1515;
}

.DEPLOYED {
  background-color: #15cb2d;
}

.ChartListItem__content__info_repo.stable {
  background-color: #1598cb;
}

.ChartListItem__content__info_repo.incubator {
  background-color: #f58220;
}
