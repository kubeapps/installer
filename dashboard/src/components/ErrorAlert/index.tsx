import ErrorAlertHeader from "./ErrorAlertHeader";
import NotFoundErrorAlert from "./NotFoundErrorAlert";
import PermissionsErrorAlert from "./PermissionsErrorAlert";
import UnexpectedErrorAlert from "./UnexpectedErrorAlert";

export { ErrorAlertHeader, NotFoundErrorAlert, PermissionsErrorAlert, UnexpectedErrorAlert };
