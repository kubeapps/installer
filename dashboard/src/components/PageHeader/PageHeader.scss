.PageHeader > .row {
  border-bottom: 2px solid #F1F1F1;
}

.PageHeader h1 {
  margin: 0;
}
