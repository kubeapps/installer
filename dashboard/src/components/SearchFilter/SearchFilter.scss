#search {
  color: var(--cds-global-typography-color-500, #000);
  margin-left: 0.3rem;
}

#search input::placeholder {
  color: var(--cds-global-typography-color-300, #454545);
}

header .search-box form {
  border-radius: 0.15rem;
  margin-left: 1rem;
}
