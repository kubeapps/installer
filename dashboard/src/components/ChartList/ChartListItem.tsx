import * as React from "react";
import { Link } from "react-router-dom";

import placeholder from "../../placeholder.png";
import { IChart } from "../../shared/types";
import Card, { CardContent, CardIcon } from "../Card";

import "./ChartListItem.css";

interface IChartListItemProps {
  chart: IChart;
}

class ChartListItem extends React.Component<IChartListItemProps> {
  public render() {
    const { chart } = this.props;
    const { icon, name, repo } = chart.attributes;
    const iconSrc = icon ? `/api/chartsvc/${icon}` : placeholder;
    const latestAppVersion = chart.relationships.latestChartVersion.data.app_version;
    return (
      <Card key={`${repo}/${name}`} responsive={true} className="ChartListItem">
        <Link to={`/charts/` + chart.id} title={name}>
          <CardIcon icon={iconSrc} />
          <CardContent>
            <div className="ChartListItem__content">
              <h3 className="ChartListItem__content__title type-big">{name}</h3>
              <div className="ChartListItem__content__info">
                <p className="ChartListItem__content__info_version margin-reset type-small padding-t-tiny type-color-light-blue">
                  {latestAppVersion || "-"}
                </p>
                <span
                  className={`ChartListItem__content__info_repo ${
                    repo.name
                  } type-small padding-t-tiny padding-h-normal`}
                >
                  {repo.name}
                </span>
              </div>
            </div>
          </CardContent>
        </Link>
      </Card>
    );
  }
}

export default ChartListItem;
