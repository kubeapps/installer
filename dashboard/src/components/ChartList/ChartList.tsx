import * as React from "react";
import { Link } from "react-router-dom";

import { IChart, IChartState } from "../../shared/types";
import { escapeRegExp } from "../../shared/utils";
import { CardGrid } from "../Card";
import { NotFoundErrorAlert } from "../ErrorAlert";
import LoadingWrapper from "../LoadingWrapper";
import PageHeader from "../PageHeader";
import SearchFilter from "../SearchFilter";
import ChartListItem from "./ChartListItem";

interface IChartListProps {
  charts: IChartState;
  repo: string;
  filter: string;
  fetchCharts: (repo: string) => Promise<{}>;
  pushSearchFilter: (filter: string) => any;
}

interface IChartListState {
  filter: string;
}

class ChartList extends React.Component<IChartListProps, IChartListState> {
  public state: IChartListState = {
    filter: "",
  };

  public componentDidMount() {
    const { repo, fetchCharts, filter } = this.props;
    this.setState({ filter });
    fetchCharts(repo);
  }

  public componentWillReceiveProps(nextProps: IChartListProps) {
    if (nextProps.filter !== this.state.filter) {
      this.setState({ filter: nextProps.filter });
    }
  }

  public render() {
    const {
      charts: { isFetching, items: allItems },
      pushSearchFilter,
    } = this.props;
    const items = this.filteredCharts(allItems, this.state.filter);
    if (!isFetching && allItems.length === 0) {
      return (
        <NotFoundErrorAlert
          resource={"Charts"}
          children={
            <div>
              Manage your Helm chart repositories in Kubeapps by visiting the{" "}
              <Link to={"/config/repos"}>App repositories configuration</Link> page.
            </div>
          }
        />
      );
    }
    const chartItems = items.map(c => <ChartListItem key={c.id} chart={c} />);
    return (
      <section className="ChartList">
        <PageHeader>
          <h1>Charts</h1>
          <SearchFilter
            className="margin-l-big"
            placeholder="search charts..."
            onChange={this.handleFilterQueryChange}
            value={this.state.filter}
            onSubmit={pushSearchFilter}
          />
        </PageHeader>
        <LoadingWrapper loaded={!isFetching}>
          <CardGrid>{chartItems}</CardGrid>
        </LoadingWrapper>
      </section>
    );
  }

  private filteredCharts(charts: IChart[], filter: string) {
    return charts.filter(c => new RegExp(escapeRegExp(filter), "i").test(c.id));
  }

  private handleFilterQueryChange = (filter: string) => {
    this.setState({
      filter,
    });
  };
}

export default ChartList;
