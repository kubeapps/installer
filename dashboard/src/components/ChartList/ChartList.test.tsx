import { shallow } from "enzyme";
import * as React from "react";

import { IChart, IChartState } from "../../shared/types";
import { NotFoundErrorAlert } from "../ErrorAlert";
import PageHeader from "../PageHeader";
import SearchFilter from "../SearchFilter";
import ChartList from "./ChartList";

const defaultChartState = {
  isFetching: false,
  selected: {} as IChartState["selected"],
  items: [],
} as IChartState;
const defaultProps = {
  charts: defaultChartState,
  repo: "stable",
  filter: "",
  fetchCharts: jest.fn(),
  pushSearchFilter: jest.fn(),
};

it("propagates the filter from the props", () => {
  const wrapper = shallow(<ChartList {...defaultProps} filter="foo" />);
  expect(wrapper.state().filter).toBe("foo");
});

describe("renderization", () => {
  it("should render an error if no charts are found", () => {
    const wrapper = shallow(<ChartList {...defaultProps} />);
    expect(wrapper.find(NotFoundErrorAlert).exists()).toBe(true);
    expect(wrapper.find(".ChartList").exists()).toBe(false);
    expect(
      wrapper
        .find(NotFoundErrorAlert)
        .children()
        .text(),
    ).toContain("Manage your Helm chart repositories");
  });

  it("should render a loading message if is still fetching charts", () => {
    const chartState = {
      isFetching: true,
      selected: {} as IChartState["selected"],
      items: [],
    } as IChartState;
    const wrapper = shallow(<ChartList {...defaultProps} charts={chartState} />);
    expect(wrapper.find(NotFoundErrorAlert).exists()).toBe(false);
    expect(wrapper.text()).toContain("Loading");
  });

  it("should render the list of charts charts", () => {
    const chartState = {
      isFetching: false,
      selected: {} as IChartState["selected"],
      items: [{ id: "foo" } as IChart, { id: "bar" } as IChart],
    } as IChartState;
    const wrapper = shallow(<ChartList {...defaultProps} charts={chartState} />);
    expect(wrapper.find(NotFoundErrorAlert).exists()).toBe(false);
    expect(wrapper.find(PageHeader).exists()).toBe(true);
    expect(wrapper.find(SearchFilter).exists()).toBe(true);
    expect(wrapper.text()).not.toContain("Loading");
    // Check list
  });
});
