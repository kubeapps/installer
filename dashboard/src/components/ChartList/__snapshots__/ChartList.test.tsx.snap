// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderization when charts available should render the list of charts 1`] = `
<section
  className="ChartList"
>
  <PageHeader>
    <h1>
      Charts
    </h1>
    <SearchFilter
      className="margin-l-big"
      onChange={[Function]}
      onSubmit={[MockFunction]}
      placeholder="search charts..."
      value=""
    />
  </PageHeader>
  <LoadingWrapper
    loaded={true}
  >
    <CardGrid>
      <ChartListItem
        chart={
          Object {
            "attributes": Object {},
            "id": "foo",
          }
        }
        key="foo"
      />
      <ChartListItem
        chart={
          Object {
            "attributes": Object {},
            "id": "bar",
          }
        }
        key="bar"
      />
    </CardGrid>
  </LoadingWrapper>
  ;
</section>
`;

exports[`renderization when fetching apps loading spinner matches the snapshot 1`] = `
<section
  className="ChartList"
>
  <PageHeader>
    <h1>
      Charts
    </h1>
    <SearchFilter
      className="margin-l-big"
      onChange={[Function]}
      onSubmit={[MockFunction]}
      placeholder="search charts..."
      value=""
    />
  </PageHeader>
  <LoadingWrapper
    loaded={false}
  >
    <CardGrid />
  </LoadingWrapper>
  ;
</section>
`;

exports[`renderization when no charts should render an error 1`] = `
<NotFoundErrorPage
  resource="Charts"
>
  <div>
    Manage your Helm chart repositories in Kubeapps by visiting the
     
    <Link
      replace={false}
      to="/config/repos"
    >
      App repositories configuration
    </Link>
     page.
  </div>
</NotFoundErrorPage>
`;
